src/common/test-webserver.c

```diff
@@ -82,6 +103,17 @@ setup (TestFixture *fixture,
                                            test_case ? test_case->server_flags : COCKPIT_WEB_SERVER_NONE);
   g_clear_object (&cert);
 
+  /* We want to check all incoming requests to ensure that they match
+   * our expectations about remote hostname and protocol.  Add a
+   * "handler" that does that, but never claims to handle anything.
+   */
+  fixture->expected_remote = g_strdup (address);
+  if (test_case && test_case->expected_protocol)
+    fixture->expected_protocol = test_case->expected_protocol;
```

Comment:
That reads really strange. If the test case overwrites the current state (fixture), what is there to check? It'd feel cleaner to not initialize it in the fixture at all, and move this check to verify_request().

Reply:
this is in `fixture_setup()`.  It should be read as initializing the value of `expected_protocol` according to the test case.